jsonargparse_tests/test_deprecated.py

```diff
@@ -155,6 +155,14 @@ def test_logger_property_none(self):
             LoggerProperty(logger=None)
 
 
+    @unittest.mock.patch.dict(os.environ, {"JSONARGPARSE_DEPRECATION_WARNINGS": "all"})
+    def test_env_prefix_none(self):
+        with warnings.catch_warnings(record=True) as w:
+            parser = ArgumentParser(env_prefix=None)
+            self.assertEquals(1, len(w))
+            self.assertIn('env_prefix', str(w[0].message))
```

Comment:
With your test I noticed that the other deprecation warnings were not really working. I did some changes, so please rebase master and look at the changed tests. Please use `str(w[-1].message)`.

Reply:
I was update to suggest in another PR the change on this, as I noticed that the logger property test was really testing anything. Thx for the refacto